### PR TITLE
patch libgit2 permissions for clone on NFS

### PR DESCRIPTION
**Pull request by pfitzseb**
Fixes https://github.com/JuliaLang/Pkg.jl/issues/1247.

### CODE REVIEW

**Comment by pfitzseb** [comment]
Not sure what to do about that FreeBSD build error, but https://stackoverflow.com/questions/46058050/unable-to-compile-unrecognized-relocation indicates that there's a binutils version mismatch.

File: L/LibGit2/bundled/patches/libgit2-nfs-permissions.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/odb_pack.c b/src/odb_pack.c
```
Comment (giordano) [inline L/LibGit2/bundled/patches/libgit2-nfs-permissions.patch:1]: Can this patch be upstreamed?

Comment (pfitzseb) [reply L/LibGit2/bundled/patches/libgit2-nfs-permissions.patch:1]: Let's see: https://github.com/libgit2/libgit2/pull/5935

**Comment by ViralBShah** [comment]
I am guessing we'll want a newer version anyways. Closing.

**Pull request closed without merging**